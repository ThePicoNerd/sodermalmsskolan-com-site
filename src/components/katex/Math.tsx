import React, { FunctionComponent } from "react";
import katex from "katex";
// eslint-disable-next-line import/extensions
import "katex/dist/contrib/mhchem.js";
import "katex/dist/katex.min.css";

export interface MathProps {
  math: string;
  display?: boolean;
}

/**
 * A component used to render KaTeX to html.
 *
 * @param {React.PropsWithChildren<MathProps>} props Props.
 *
 * @returns {React.ReactElement} The rendered math.
 */
const Math: FunctionComponent<MathProps> = ({ math, display = false }) => {
  const html = katex.renderToString(math, {
    displayMode: display,
    output: "html",
  });

  return (
    <span
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{
        __html: html,
      }}
    />
  );
};

export default Math;
