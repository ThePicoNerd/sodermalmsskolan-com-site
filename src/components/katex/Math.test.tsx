import React from "react";
import renderer from "react-test-renderer";
import Math from "./Math";

describe("<Math /> test", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Math math="E=mc^2" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
