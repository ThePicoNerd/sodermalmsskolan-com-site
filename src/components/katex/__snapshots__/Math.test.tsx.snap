// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Math /> test renders correctly 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "<span class=\\"katex\\"><span class=\\"katex-html\\" aria-hidden=\\"true\\"><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.68333em;vertical-align:0em;\\"></span><span class=\\"mord mathnormal\\" style=\\"margin-right:0.05764em;\\">E</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span><span class=\\"mrel\\">=</span><span class=\\"mspace\\" style=\\"margin-right:0.2777777777777778em;\\"></span></span><span class=\\"base\\"><span class=\\"strut\\" style=\\"height:0.8141079999999999em;vertical-align:0em;\\"></span><span class=\\"mord mathnormal\\">m</span><span class=\\"mord\\"><span class=\\"mord mathnormal\\">c</span><span class=\\"msupsub\\"><span class=\\"vlist-t\\"><span class=\\"vlist-r\\"><span class=\\"vlist\\" style=\\"height:0.8141079999999999em;\\"><span style=\\"top:-3.063em;margin-right:0.05em;\\"><span class=\\"pstrut\\" style=\\"height:2.7em;\\"></span><span class=\\"sizing reset-size6 size3 mtight\\"><span class=\\"mord mtight\\">2</span></span></span></span></span></span></span></span></span></span></span>",
    }
  }
/>
`;
